lib/story_query.rb

```diff
@@ -1,34 +1,30 @@
 require_dependency 'user_query'
 
 class StoryQuery
-
   # Returns a list of stories having the given IDs.
   def self.find_by_ids(story_ids, current_user)
-    stories = Story.where(id: story_ids).for_user(current_user)
-                   .includes(:user, :substories, :target,
-                             substories: [:user, :target])
+    stories = Story.where(id: story_ids)
+              .for_user(current_user)
+              .includes(:user, :substories, :target,
+                        substories: %i(user target))
 
     # Preload genres for anime stories.
-    anime_stories = stories.select {|s| s.target_type == "Anime" }.map(&:target)
-                   .index_by(&:id)
-    Anime.where(id: anime_stories.values.map(&:id)).includes(:genres).each do |a|
-      assoc = anime_stories[a.id].association(:genres)
-      assoc.loaded!
-      assoc.target.concat(a.genres)
-      a.genres.each {|genre| assoc.set_inverse_instance(genre) }
-    end
+    stories.select { |s| s.target_type == 'Anime' }.map(&:target).index_by(&:id)
```

Comment:
The loading seems fine:
```
Story Load (1.2ms)  SELECT  "stories"."id" FROM "stories" LEFT OUTER JOIN watchlists ON watchlists.id = stories.watchlist_id WHERE "stories"."deleted_at" IS NULL AND "stories"."user_id" = $1 AND (NOT adult) AND (watchlists.id IS NULL OR watchlists.private = 'f') AND (stories.group_id IS NULL)  ORDER BY stories.updated_at DESC LIMIT 30 OFFSET 0  [["user_id", 1]]
Story Load (2.7ms)  SELECT "stories".* FROM "stories" LEFT OUTER JOIN watchlists ON watchlists.id = stories.watchlist_id WHERE "stories"."deleted_at" IS NULL AND "stories"."id" IN (1, 2) AND (NOT adult) AND (watchlists.id IS NULL OR watchlists.private = 'f')
Substory Load (2.4ms)  SELECT "substories".* FROM "substories"  WHERE "substories"."deleted_at" IS NULL AND "substories"."story_id" IN (2, 1)
Anime Load (1.7ms)  SELECT "anime".* FROM "anime"  WHERE "anime"."id" IN (3936)
HABTM_Genres Load (1.8ms)  SELECT "anime_genres".* FROM "anime_genres"  WHERE "anime_genres"."anime_id" IN (3936)
Genre Load (2.8ms)  SELECT "genres".* FROM "genres"  WHERE "genres"."id" IN (28, 8, 11, 4, 2, 1)
```
I've added all the possibilities I know that would break things, and everything seems ok. If you remember some edge case, lmk